js/authorization.js

```diff
@@ -33,159 +38,190 @@ const validateEmail = (target) => {
-      type: target.type,
-      target,
-    });
-  } else {
-    removeErrorMsg(target);
+  const email = { email: target.value };
+
+  try {
+    const checkEmailResponse = await requestAPI("check-email", email);
+    if (checkEmailResponse.status === 409) {
+      throw new Error("existingEmail");
+    }
+    removeErrorMessage(target);
+  } catch (error) {
+    if (error.message === "existingEmail") {
+      paintErrorMessage({
+        error: "unavailableEmail",
+        type: target.type,
+        target,
+      });
+    } else {
+      alert(error.message);
+    }
   }
 };
 
 const checkPasswordMatch = () => {
-  if (!$authPassword.value || !$authPasswordCheck.value) {
+  if (!authPassword.value || !authPasswordCheck.value) {
     return;
   }
 
-  if ($authPassword.value !== $authPasswordCheck.value) {
-    paintErrorMsg({
+  if (authPassword.value !== authPasswordCheck.value) {
+    paintErrorMessage({
       error: "invalidInput",
-      type: "password-check",
-      target: $authPasswordCheck,
+      type: "passwordCheck",
+      target: authPasswordCheck,
     });
   } else {
-    removeErrorMsg($authPasswordCheck);
+    removeErrorMessage(authPasswordCheck);
   }
 };
 
 const togglePasswordVisibility = (event) => {
-  event.preventDefault();
-
-  const $passwordInput =
+  const passwordInput =
     event.currentTarget.parentElement.querySelector(".form__input-box");
-  const $togglePasswordImg = event.currentTarget.parentElement.querySelector(
+  const togglePasswordImg = event.currentTarget.parentElement.querySelector(
     ".form__password-toggle img"
   );
 
-  if ($passwordInput.type === "password") {
-    $passwordInput.type = "text";
-    $togglePasswordImg.src = "/public/icons/eye-on.svg";
+  if (passwordInput.type === "password") {
+    passwordInput.type = "text";
+    togglePasswordImg.src = "/public/icons/eye-on.svg";
   } else {
-    $passwordInput.type = "password";
-    $togglePasswordImg.src = "/public/icons/eye-off.svg";
+    passwordInput.type = "password";
+    togglePasswordImg.src = "/public/icons/eye-off.svg";
   }
 };
 
-const handleSigninSubmit = (event) => {
+const redirect = () => {
+  location.href = "/pages/folder.html";
+};
+
+const handleSigninSubmit = async (event) => {
   event.preventDefault();
 
-  if (
-    $authEmail.value === ACCOUNT.email &&
-    $authPassword.value === ACCOUNT.password
-  ) {
-    location.href = "/pages/folder.html";
-  } else {
-    paintErrorMsg({ error: "invalidLogin", type: "email", target: $authEmail });
-    paintErrorMsg({
-      error: "invalidLogin",
-      type: "password",
-      target: $authPassword,
-    });
+  const account = { email: authEmail.value, password: authPassword.value };
+
+  try {
+    const signinResponse = await requestAPI("sign-in", account);
+    if (signinResponse.status === 400) {
+      throw new Error("invalidCredentials");
+    }
+    const signinData = await signinResponse.json();
+    const accessToken = signinData.data.accessToken;
```

Comment:
저는 아래와 같이 바로 구조분해할당을 사용하기도 해요! 

```suggestion
    const { data: { accessToken } } = signinData;
```

Reply:
좋아요! 참고로 커밋해시는 아래와 같이 공유해주시면 됩니다.
https://github.com/codeit-bootcamp-frontend/1-Weekly-Mission/pull/255/commits/546da48dfc60e079aa64ced100185cabccc3f9e1

이렇게 공유하면 링크로 바로 이동할 수 있어요!

Reply:
앗 그렇군요! 앞으로는 알려주신 방식으로 공유하도록 하겠습니다!